columnflow/tasks/framework/base.py

```diff
@@ -266,6 +266,197 @@ def has_obj(name):
+                    if inference_model in (None, "NO_STR"):
+                        inference_model = container.x.default_inference_model
+
+                    # get the default_ml_model from the inference_model_inst
+                    inference_model_inst = columnflow.inference.InferenceModel._subclasses[inference_model]
+                    default_ml_model = getattr(inference_model_inst, "ml_model_name", default_ml_model)
+
+                    return default_ml_model
+
+                return default_ml_model
+
+            config_inst = od.Config(
+                id=0,
+                name="my_config",
+                aux={"default_ml_model": default_ml_model},
+            )
+
+            @inference_model(ml_model_name="default_ml_model")
+            def my_inference_model(self):
+                # some inference model implementation
+                ...
+
+            params = {"config_inst": config_inst, "ml_model": None, "inference_model": "my_inference_model"}
+            resolve_param_values(params)  # sets params["ml_model"] to "my_ml_model"
+
+            params = {"config_inst": config_inst, "ml_model": "some_ml_model", "inference_model": "my_inference_model"}
+            resolve_param_values(params)  # sets params["ml_model"] to "some_ml_model"
+        """
+
+        # get the container inst
+        if isinstance(container, str):
+            container = task_params.get(container, None)
+
+        if not container:
+            return param
+
+        # expand default when param is empty
+        if param in (None, law.NO_STR, tuple()):
+            param = container.x(default_str, None) if default_str else None
+
+            # allow default to be a function, taking task parameters as input
+            if isinstance(param, Callable):
+                param = param(container, **task_params)
+
+        # return either a tuple or only the first param, based on the *multiple* parameter
+        if multiple:
+            return law.util.make_tuple(param, cast=True)
+        elif param and isinstance(param, (list, tuple)):
+            return param[0]
+
+        return param
+
+    @classmethod
+    def resolve_config_default_and_groups(
+        cls,
+        task_params: dict[str, Any],
+        param: str | tuple[str] | None,
+        container: str | od.AuxDataMixin = "config_inst",
+        default_str: str | None = None,
+        groups_str: str | None = None,
+    ) -> tuple[str]:
+        """
+        Resolves tuple of inputs *param* by using groups identified via the *groups_str* from
+        an :py:class:`order.AuxDataMixin` *container* that maps a string to a tuple of strings.
+        If *param* is empty, the default identified by *default_str* from the *container* is returned.
+        If *default_str* points to a function, it should take *container* and *task_params* as an input.
+
+        Example:
+
+        .. code-block:: python
+            def resolve_param_values(params):
```

Comment:
Can you simplify this example? I tried to recap what the method does, but the example code is a bit entangled. Like

```python
config_inst = ...

params = {"producer": None}

ConfigTask.resolve_config_default_and_groups(
    params,
    params.get("producer"),
    container=config_inst,
    default_str="default_producer",
    groups_str="producer_groups",
)
# -> ("features_1", "features_2", "features_3")
```